package/MDAnalysis/analysis/lineardensity.py

```diff
@@ -0,0 +1,260 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding:utf-8 -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4 
+#
+# MDAnalysis --- http://www.MDAnalysis.org
+# Copyright (c) 2006-2015 Naveen Michaud-Agrawal, Elizabeth J. Denning, Oliver Beckstein
+# and contributors (see AUTHORS for the full list)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+
+"""
+Linear Density --- :mod:`MDAnalysis.analysis.lineardensity`
+===========================================================
+
+A tool to compute mass and charge density profiles along the three
+cartesian axes of the simulation cell. Works only for orthorombic,
+fixed volume cells (thus for simulations in canonical NVT ensemble).
+"""
+from __future__ import division
+
+import os.path as path
+import numpy as np
+
+from MDAnalysis.analysis.base import AnalysisBase
+
+class LinearDensity(AnalysisBase):
+    """Linear density profile
+    LinearDensity(universe, selection, grouping='atoms', binsize=0.25)
+
+    Parameters
+    ----------
+    universe : `Universe` object
+      universe whose density profiles will be computed
+    selection : `AtomGroup` object
+      Any atomgroup belonging to universe
+
+    Keywords
+    --------
+    grouping : str {'atoms', 'residues', 'segments', 'fragments'}
+          Density profiles will be computed on the center of geometry
+          of a selected group of atoms ['atoms']
+
+    binsize : float
+          Bin width in Angstrom used to build linear density
+          histograms. Defines the resolution of the resulting density 
+          profile (smaller --> higher resolution) [0.25]
+
+    start : int
+          The frame to start at [0]
+    stop : int
+          The frame to end at [-1]
+    step : int
+          The step size through the trajectory in frames [0]
+
+    Example
+    -------
+    First create a LinearDensity object by supplying a universe and
+    a selection, then use the `run` method
+      ldens = LinearDensity(universe, selection)
+      ldens.run()
+    Analysis can be run in parallel on multicore machines by supplying
+    the argument `parallel=True` to the `run` method:
+      ldens.run(parallel=True, nthreads=4)
+    
+    Density proiles are output to file through the `save` method:
+      ldens.save(description='mydensprof', form='txt')
+    which will write the density profiles in a file named
+    <trajectory_filename>.mydensprof_<grouping>.ldens
+    Results can be output in npz format by specifying `form='npz'`
+
+    .. versionadded:: 0.14.0
+    """
+    def __init__(self, universe, selection, grouping='atoms', binsize=0.25,
```

Comment:
You don't need to ask for Universe here, as it's always available as `selection.universe`